### DISCUSSION TITLE
Tasmota 9.1.0 -> 9.3.0 failing

### DISCUSSION THREAD

**Discussion by daniel-lo** [arendst/Tasmota#11641] [Q&A] [answered]
After flashing tasmota.bin 9.1.0 to my devices I cannot flash 9.3.0 tasmota-minimal.bin.

Is this by design? I'd like to update.

**Comment by barbudor** [comment, marked as answer]
Can you try to flash through OTA URL using http://ota.tasmota.com/tasmota/release/tasmota.bin.gz
All versions after 8.3.1 supports compressed firmware and it is the recommended method since 9.1.0
In most of the cases (such as tasmota.bin.gz) the upgrade will go directly without going through minimal due to the smaller size of the compressed binary.
If you were trying to upgrade with a larger firmware, big enough to require going through minimal, using OTA URL (instead of file upload) would also perform the 2 steps (minimal then designated firmware) automatically.

If you can't use OTA URL and must use file upload, try uploading tasmota-lite.bin.gz *(always **.gz***) instead
tasmota-lite is recommended when performing an upgrade path because it will convert the old settings in the flash into new format while tasmota-minimal doesn't.

**Comment by daniel-lo** [comment]
Thanks .. the OTA URL worked!

**Comment by daniel-lo** [comment]
Need to find out how to do this with Tasmoadmin

**Reply by barbudor** [reply]
You need to find out  how to get rid of TasmoAdmin.
This software makes continuous aggressive HTTP requests to tasmota devices which push them to the limit and creates more problems than it solves.
Check [TDM](https://github.com/jziolkowski/tdm) or the new work-in-progress [TasmoManager](https://github.com/arendst/Tasmota/discussions/11509)

**Comment by daniel-lo** [comment]
Will do .. thanks
